source/common/http/conn_manager_impl.h

```diff
@@ -691,6 +695,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
   const Server::OverloadActionState& overload_stop_accepting_requests_ref_;
   const Server::OverloadActionState& overload_disable_keepalive_ref_;
   TimeSource& time_source_;
+  absl::flat_hash_map<const std::string, PerConnectionObjectSharedPtr> per_connection_object_map_;
```

Comment:
Perf nit: Can you lazy create this map as it won't be used in almost all cases? See what we do for `saved_request_metadata_` for example.

Reply:
Thanks, this was interesting. I was thinking about using static references, did not get the desired results; but this method is more effective.